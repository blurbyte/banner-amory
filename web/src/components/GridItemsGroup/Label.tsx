import styled from 'styled-components';

import Headline from '../Headline';

const Paragraph = Headline.withComponent('p');

const Label = styled(Paragraph)`
  text-transform: uppercase;
  margin: 0;
`;

export default Label;
