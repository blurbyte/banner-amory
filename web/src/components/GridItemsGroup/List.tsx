// grid containing items under single character

import styled from 'styled-components';

type ListProps = {
  columns: number;
};

const List = styled.ul<ListProps>`
  display: grid;
  grid-template-columns: repeat(${props => props.columns}, 1fr);
  grid-gap: 1rem;
  margin: 0;
  padding: 0;
  list-style: none;
`;

export default List;
