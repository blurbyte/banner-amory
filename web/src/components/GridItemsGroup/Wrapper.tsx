import styled, { keyframes } from 'styled-components';

const slideIn = keyframes`
  from {
    transform: translateY(-1.2rem);
    opacity: 0;
  }
  to {
    transform: translateY(0);
    opacity: 1;
  }
`;

type WrapperProps = {
  columns: number;
  animationDelay: number;
};

const Wrapper = styled.div<WrapperProps>`
  display: grid;
  grid-column: span ${props => props.columns};
  opacity: 0;
  animation: ${slideIn} 300ms ease-in forwards ${props => props.animationDelay}ms;
`;

export default Wrapper;
