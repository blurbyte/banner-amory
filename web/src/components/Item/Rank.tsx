import styled from 'styled-components';

import { Colors } from '../../styles/colors';

const Rank = styled.p`
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  right: 0;
  bottom: 0;
  z-index: 99;
  background: ${Colors.white};
  margin: 0;
  padding: 0;
  width: 2rem;
  height: 2rem;
  font-weight: 600;
`;

export default Rank;
