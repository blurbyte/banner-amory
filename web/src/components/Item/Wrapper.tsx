import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '../../styles/colors';

const ListItem = styled.li`
  position: relative;
  background: ${transparentize(0.9, Colors.blue)};
  z-index: 99;
`;

export default ListItem;
