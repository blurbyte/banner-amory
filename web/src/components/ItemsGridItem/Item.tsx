// item thumb wrapper
// modifies the look of overlay if it is last element in row
// or the last element of the grid

import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '@styles/colors';
import Overaly from './Overlay';

const Item = styled.li`
  background: ${transparentize(0.9, Colors.blue)};
  user-select: none;

  &:nth-child(10n) ${Overaly}, &:last-child:not(:first-child) ${Overaly} {
    left: auto;
    right: 0;
    justify-content: flex-start;
  }
`;

export default Item;
