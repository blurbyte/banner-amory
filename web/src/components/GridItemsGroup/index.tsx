// items grouped by character

import * as React from 'react';

import { ItemBasic } from '@sharedTypes/Item';
import GridItem from '../ItemsGridItem';
import Wrapper from './Wrapper';
import Label from './Label';
import List from './List';

const MAX_ITEMS_PER_ROW = 10; // number of columns in base Grid

type GridItemsGroupProps = {
  label: string;
  items: ItemBasic[];
  key?: string;
};

class GridItemsGroup extends React.Component<GridItemsGroupProps> {
  render() {
    const { label, items } = this.props;

    const numberOfItems = items.length + 1; // add 1 to make 1 column space betweeen groups
    const maxColumns = numberOfItems > MAX_ITEMS_PER_ROW ? MAX_ITEMS_PER_ROW : numberOfItems;

    return (
      <Wrapper columns={maxColumns} animationDelay={200}>
        <Label>{label}</Label>
        <List columns={maxColumns}>
          {items.map(item => <GridItem key={item.slug} {...item} />)}
        </List>
      </Wrapper>
    );
  }
}

export default GridItemsGroup;
