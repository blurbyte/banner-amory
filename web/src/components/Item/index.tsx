import * as React from 'react';

import Wrapper from './Wrapper';
import Image from './Image';
import Rank from './Rank';

class Item extends React.Component {
  render() {
    return (
      <Wrapper>
        <Image src={`${process.env.STORAGE_URL}alettes-bracelet.png`} />
        <Rank>5</Rank>
      </Wrapper>
    );
  }
}

export default Item;
