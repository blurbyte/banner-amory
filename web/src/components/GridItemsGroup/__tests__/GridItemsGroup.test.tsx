import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { ItemBasic } from '@sharedTypes/Item';
import GridItemsGroup from '../index';

test('renders correctly', () => {
  type Props = {
    label: string;
    items: ItemBasic[];
  };

  const props: Props = {
    label: 'x',
    items: [
      {
        name: 'Xanathous Crown',
        slug: 'xanathous-crown',
        gamePart: 1,
        rank: 4
      },
      {
        name: 'Xyndra Grimore',
        slug: 'xyndra-grimore',
        gamePart: 2,
        rank: 6
      }
    ]
  };

  const component = renderer.create(<GridItemsGroup {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
