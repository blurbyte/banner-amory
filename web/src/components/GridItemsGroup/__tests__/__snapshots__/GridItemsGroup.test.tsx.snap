// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-VigVT cReqnd"
>
  <p
    className="sc-jzJRlG cyaFDf sc-fjdhpX dFoqyJ"
  >
    x
  </p>
  <ul
    className="sc-cSHVUG cvBfsK"
  >
    <li
      className="sc-bxivhb fuAhs"
    >
      <a
        className="sc-ifAKCX GbWcO"
        href="/item/xanathous-crown"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-bdVaJa dMhIUw"
        >
          <img
            className="sc-bwzfXH iIjYES"
            src="undefinedxanathous-crown.png"
          />
        </div>
        <p
          className="sc-gZMcBi hwidfF sc-bZQynM geAaEl"
        >
          4
        </p>
      </a>
    </li>
    <li
      className="sc-bxivhb fuAhs"
    >
      <a
        className="sc-ifAKCX GbWcO"
        href="/item/xyndra-grimore"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-bdVaJa dMhIUw"
        >
          <img
            className="sc-bwzfXH iIjYES"
            src="undefinedxyndra-grimore.png"
          />
        </div>
        <p
          className="sc-gZMcBi hwidfF sc-bZQynM geAaEl"
        >
          6
        </p>
      </a>
    </li>
  </ul>
</div>
`;
